api/system/examples/update-state-in-serial-message-callback.rs

```diff
@@ -0,0 +1,68 @@
+#![no_std]
+extern crate alloc;
+
+#[macro_use]
+extern crate sys;
+extern crate playdate_system as system;
+
+use alloc::string::String;
+use core::ptr::NonNull;
+
+use sys::EventLoopCtrl;
+use sys::ffi::*;
+use system::System;
+use system::event::SystemEventExt as _;
+use system::prelude::*;
+
+struct State {
+	latest_message: Option<String>,
+}
+
+impl State {
+	fn new() -> Self { Self { latest_message: None } }
+
+	/// System event handler
+	fn event(&'static mut self, event: SystemEvent) -> EventLoopCtrl {
+		match event {
+			SystemEvent::Init => {
+				System::Default().set_serial_message_callback(Some(|msg| {
+					                 self.latest_message = Some(msg);
+				                 }));
+
+				println!("Game init complete");
+			},
+			_ => {},
+		}
+		EventLoopCtrl::Continue
+	}
+}
+
+impl Update for State {
+	fn update(&mut self) -> UpdateCtrl {
+		if let Some(latest_message) = self.latest_message.clone() {
+			println!("Latest message: {:#?}", latest_message);
+			self.latest_message = None;
+		}
+		UpdateCtrl::Continue
+	}
+}
+
+/// Entry point
+#[no_mangle]
+pub fn event_handler(_api: NonNull<PlaydateAPI>, event: SystemEvent, _sim_key_code: u32) -> EventLoopCtrl {
+	pub static mut STATE: Option<State> = None;
+
+	if unsafe { STATE.is_none() } {
+		let state = State::new();
+		unsafe { STATE = Some(state) }
+	}
+
+	if event == SystemEvent::Init {
+		unsafe { STATE.as_mut().expect("impossible").set_update_handler() }
+	}
+
+	unsafe { STATE.as_mut().expect("impossible").event(event) }
```

Comment:
I think it’s safe to do this because the previous mutable borrow doesn’t actually mutate anything when calling `set_update_handler`. I could be wrong about that, and it could be an implementation detail that changes in the future.

It would be nice if we could use `thread_local!`, but that doesn’t work in `no_std` environments. An API similar to the one it provides (closure-based access) might provide similar benefits.

Reply:
This is safe because `static` must not be moved __and__ we just ensure and set it on [previous line](https://github.com/boozook/playdate/pull/375/files/090aecfb7db9e3743ff438684e428f1610844a79#diff-9088546d7f194b571863d3247b63ee540292c3bc96ffe0a190e70a162a94239fR56), so that's ok 👍🏻.